connection.go

```diff
@@ -502,6 +505,13 @@ func (conn *Connection) dial() (err error) {
 	conn.Greeting.Version = bytes.NewBuffer(greeting[:64]).String()
 	conn.Greeting.auth = bytes.NewBuffer(greeting[64:108]).String()
 
+	// IPROTO_ID requests can be processed without authentication.
+	// https://www.tarantool.io/en/doc/latest/dev_guide/internals/iproto/requests/#iproto-id
+	if err = conn.loadProtocolInfo(w, r); err != nil {
+		connection.Close()
+		return err
+	}
+
```

Comment:
What about https://github.com/tarantool/go-tarantool/pull/226#issuecomment-1312781708 ?

Reply:
Implemented in new version